### ISSUE TITLE
Use a secure GnuPG module

### ISSUE THREAD

**Issue by Hainish** [freedomofpress/securedrop#250]
The current module we are using (PyPI: python-gnupg) is vulnerable to remote code execution via specially crafted client-provided messages due "to unsanitised inputs being passed to subprocess.Popen([...], shell=True)"

A secure forked implementation has been developed. (PyPI: gnupg)  https://github.com/isislovecruft/python-gnupg

Some of the methods we are using in python-gnupg were not implemented in the secured fork.  I've re-implemented them: https://github.com/isislovecruft/python-gnupg/pull/40

Once this pull request is merged and the PyPI module is updated, we should change our requirements txts to use this module rather than the insecure one.

**Comment by garrettr** [comment]
We are currently using the fork of python-gnupg 1.2.5 that @Hainish created. This is the "secure" python-gnupg module meant to protect from RCE via shell injection. We had to use the fork because several changes in python-gnupg 1.3 broke functionality that was required by SecureDrop.

I developed fixes for these issues (https://github.com/isislovecruft/python-gnupg/pull/18, https://github.com/isislovecruft/python-gnupg/pull/92), which were recently merged into python-gnupg 2.0.0. As a result, I am going to:

1. Close this issue, since we are using a "secure" python-gnupg module
2. Open a new issue to switch back to using the stable release of python-gnupg, and discontinue using the fork
